src/main/scala/com/gradle/SbtCommonCustomUserDataPlugin.scala

```diff
@@ -0,0 +1,44 @@
+package com.gradle
+
+import com.gradle.develocity.agent.sbt.DevelocityPlugin
+import com.gradle.develocity.agent.sbt.DevelocityPlugin.autoImport.DevelocityConfiguration
+import com.gradle.internal.{CustomBuildScanConfig, CustomBuildScanEnhancements, CustomServerConfig, Overrides}
+import sbt.Keys._
+import sbt._
+
+object SbtCommonCustomUserDataPlugin extends AutoPlugin {
+
+  override def requires: Plugins = com.gradle.develocity.agent.sbt.DevelocityPlugin
+
+  // This plugin is automatically enabled for projects which have DevelocityPlugin.
+  override def trigger = allRequirements
+
+  override lazy val buildSettings: Seq[Setting[_]] = Seq(
+    DevelocityPlugin.autoImport.develocityConfiguration := applyCCUD(
+      DevelocityPlugin.autoImport.develocityConfiguration.value,
+      scalaVersion.value)
+  )
+
+  private def applyCCUD(currentConfiguration: DevelocityConfiguration, scalaVersion: String): DevelocityConfiguration = {
+    val scan = currentConfiguration.buildScan
+    val server = currentConfiguration.server
+
+    val customServerConfig = new CustomServerConfig().fromServer(server)
+    val customBuildScanConfig = new CustomBuildScanConfig()
+    customBuildScanConfig.addValue("Scala version", scalaVersion)
+
+    new Overrides(customServerConfig).apply()
+    new CustomBuildScanEnhancements(customBuildScanConfig, customServerConfig).apply()
```

Comment:
My personal preference would be to avoid having a mutable object that different classes pass around and mutate (`customBuildScanConfig` here). Instead, I'd prefer to have `CustomBuildScanEnhancements` maintain a private mutable state while collecting all the additional links, values etc.

Something like this:
```scala
class CustomBuildScanEnhancements {
  private val config = new CustomBuildScanConfig()

  def withAdditionalData(buildScan: BuildScan): BuildScan = {
    captureOs()
    captureIDE()
    // ...
    buildScan.withTags(config.tags)
             .withValues(config.values)
             .withLinks(config.links)
  }
```